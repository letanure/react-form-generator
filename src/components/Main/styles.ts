import styled from 'styled-components'

export const Wrapper = styled.main`
  background-color: #ffffff;
  width: 96vw;
  max-width: 1200px;
  margin: 2vw auto;
  box-shadow: 0 1px 8px rgb(0 0 0 / 8%);
  height: 100%;
  padding: 3rem;
  display: flex;
  flex-direction: column;
  border-radius: 8px;
  display: flex;
  flex-direction: row;
  align-items: flex-start;
`

export const FormArea = styled.div`
  width: 33.33%;
  padding-right: 30px;
`
export const CodeArea = styled.div`
  width: 66.66%;
  position: sticky;
  top: 30px;
  display: flex;
  & > div {
    width: 50%;
    padding: 15px;
    overflow: hidden;
    pre {
      overflow: hidden;
    }
  }
`

export const Description = styled.h2`
  font-size: 2rem;
  font-weight: 400;
`
