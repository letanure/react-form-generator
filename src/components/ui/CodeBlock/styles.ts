import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${() => css`
    font-size: 11px;
    border: 1px solid #999;
    padding: 10px;
    line-height: 1.2em;
  `}
`
