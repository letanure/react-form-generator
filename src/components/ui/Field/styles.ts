import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${() => css`
    padding-bottom: 15px;
    &.field-hidden {
      display: none;
    }
    .label {
      visibility: inherit;
      margin: 6px 0px;
      font-size: 12px;
      font-weight: 600;
      color: #0c072c;
      display: block;
    }
    textarea {
      font-family: inherit;
      color: #0c072c;
      background: #fff;
      width: 100%;
      padding: 8px;
      font-size: 14px;
      border: 1px solid #0c072c33;
      border-radius: 8px;
      min-height: 70px;
      display: block;

      &:focus,
      &:focus-visible {
        outline: 1px solid #b336a0;
        border-color: #b336a0;
      }

      &.hasError {
        border: 1px solid rgb(233, 75, 75);
      }
    }
    input {
      font-family: inherit;
      color: #0c072c;
      background: #fff;
      width: 100%;
      padding: 12px;
      font-size: 14px;
      border: 1px solid #0c072c33;
      border-radius: 8px;

      &:focus,
      &:focus-visible {
        outline: 1px solid #b336a0;
        border-color: #b336a0;
      }

      &.hasError {
        border: 1px solid rgb(233, 75, 75);
      }

      &[type='color'] {
        width: 50px;
        padding: 4px;
        height: 43px;
      }
    }
  `}
`

export const ErrorMessage = styled.div`
  padding-top: 4px;
  font-size: 14px;
  color: rgb(233, 75, 75);
`
