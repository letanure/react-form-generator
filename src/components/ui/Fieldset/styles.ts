import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${() => css`
    margin-bottom: 10px;
  `}
`
