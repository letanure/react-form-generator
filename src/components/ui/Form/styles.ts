import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${() => css`
    padding: 15px;
  `}
`

export const Title = styled.h2`
  ${() => css`
    font-size: 4rem;
    margin-bottom: 15px;
  `}
`

export const Description = styled.div`
  ${() => css`
    font-size: 2rem;
    margin-bottom: 15px;
  `}
`

export const Button = styled.button`
  ${() => css`
    font-family: inherit;
    padding: 12px 24px;
    margin-left: auto;
    font-size: 16px;
    font-weight: 500;
    color: rgb(255, 255, 255);
    cursor: pointer;
    background-color: #b336a0;
    background-size: 50px 50px;
    border: none;
    border-radius: 8px;

    &:hover {
      background-color: #b336a0cc;
    }
  `}
`
