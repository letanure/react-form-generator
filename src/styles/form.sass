.form
  --color-primary: #2e3440
  --color-inactive: #d8dee9
  --color-hover: #afb4bd
  --color-active: #5e81ac
  --color-error: #BF616A
  --target-height: 40px
  --font-size: calc(var(--target-height) * 0.4)
  --height: calc(var(--font-size) * 2.5)
  --base-padding: calc((var(--height) - var(--font-size)) / 2)
  --border-radius: calc(var(--font-size) / 4)
  --background-color: transparent
  --border-width: calc(var(--font-size) / 9)
  --border-color: var(--color-inactive)
  --border-style: solid
  --border-inactive: var(--border-width) var(--border-style) var(--border-color)
  --border-active: var(--border-width) var(--border-style) var(--color-active)
  --border-error: var(--border-width) var(--border-style) var(--color-error)
  --border: var(--border-inactive)
  --line-height: calc(var(--target-height) - (var(--border-width) * 2))
  --label-fz: calc(var(--font-size) * 0.813)
  --label-fw: 600
  --label-c: var(--color-primary)
  --label-pb: calc(var(--base-padding) / 2)
  --transition-field: color 0.1s ease, border-color 0.1s ease

  box-sizing: border-box
  font-size: var(--font-size)

  fieldset, legend
    border: none
    padding: 0

  .fieldset
    grid-area: fieldset
    display: grid

  input:not([type='checkbox'], [type='radio']), textarea, select
    appearance: none
    background-color: var(--background-color)
    border-radius: var(--border-radius)
    border: var(--border)
    border-color: var(--border-color)
    font-size: var(--font-size)
    height: var(--height)
    padding: 0 var(--base-padding)
    transition: var(--transition-field)
    width: 100%
    &:hover,
    &:focus,
    &:focus-visible
      outline: none
      // --border-color: var(--color-hover)

  .field
    accent-color: var(--color-active)
    display: grid
    grid-template-areas: "label" "control" "errors"
    grid-template-columns: 1fr
    grid-template-rows: auto
    margin-bottom: var(--base-padding)
    transition: var(--transition-field)

    *
      box-sizing: border-box
      padding: 0

    &[data-valid="false"]
      --border-color: var(--color-error)
    &[data-valid="true"]
      --border-color: var(--color-inactive)
    &[data-focus="true"], &[data-valid="false"][data-focus="true"]
      --border-color: var(--color-active)
    // &[data-hover="true"]
    &:hover
      --border-color: var(--color-hover)

    &.layout-horizontal, &.layout-horizontal-bordered
      grid-template-areas: "label control" ".  errors "
      grid-template-columns: 1fr 2fr
      .field-label
        padding-bottom: 0
      legend
        display: contents
        line-height: var(--line-height)
      &.field-array > .field-label
        align-items: start
        line-height: var(--line-height)
      &.field-object .field
        &:nth-child(1)::before
          height: 0
          top: calc(var(--height) / 2)
        &::before
          width: calc(var(--base-padding) *0.7)

    &.layout-horizontal-bordered
      grid-template-columns: auto 2fr
      grid-template-areas: "label control" "errors errors "
      .field-label
        border: var(--border-inactive)
        border-color: var(--border-color)
        border-right: 0
        border-top-left-radius: var(--border-radius)
        border-bottom-left-radius: var(--border-radius)
        padding-left: var(--base-padding)
      &.field-radioGroup
        border: var(--border-inactive)
        padding-left: var(--base-padding)
        border-radius: var(--border-radius)
        height: var(--target-height)
        legendXX
          display: block
      &.field-select
        select
          border-left: 0
          border-top-left-radius: 0
          border-bottom-left-radius: 0
      &.field-textarea
        grid-template-areas: "label label" "control control" "errors errors"
        .field-label
          border-bottom: 0
          border-top-right-radius: var(--border-radius)
          border-bottom-left-radius: 0
          border-right: var(--border-inactive)
          padding-top: calc(var(--base-padding) / 2)
          border-color: var(--border-color)
        textarea
          border-top: 0
          border-top-left-radius: 0
          border-top-right-radius: 0
          padding: calc(var(--base-padding) / 2) var(--base-padding)
      &.field-range
        .field-control
          border-bottom-right-radius: var(--border-radius)
          border-top-right-radius: var(--border-radius)
          border: var(--border-inactive)
          border-left: 0
          border-color: var(--border-color)
          height: var(--target-height)
          padding-right: var(--base-padding)
      &.field-object
        grid-template-areas: "label label" "control control" "errors errors"
        padding: 0 var(--base-padding)
        border: var(--border-inactive)
        border-radius: var(--border-radius)
        border-color: var(--border-color)
        legend
          line-height: var(--line-height)
        .field::before
          width: var(--base-padding)
      &.field-array
        grid-template-areas: "label label" "control control" "errors errors"
        padding: 0 var(--base-padding)
        border: var(--border-inactive)
        border-radius: var(--border-radius)
        border-color: var(--border-color)
        li
          grid-template-areas: "fieldset fieldset fieldset" "remove remove remove"
          border: var(--border-inactive)
          padding: var(--base-padding)
          border-radius: var(--border-radius)
          border-color: var(--border-color)
          &:before
            margin-left: calc( (var(--base-padding) * -3) - var(--border-width))
        & > .field-label
          border: 0
          padding: 0
      input
        border-left: 0
        border-top-left-radius: 0
        border-bottom-left-radius: 0



    &-label
      align-items: center
      color: var(--label-c)
      display: flex
      font-size: var(--label-fz)
      font-weight: var(--label-fw)
      grid-area: label
      padding-bottom: var(--label-pb)
      white-space: nowrap

    &-control
      grid-area: control
      min-width: 0
      position: relative

    &-remove
      grid-area: remove

    &-errors
      color: var(--color-error)
      font-size: var(--label-fz)
      grid-area: errors

    &-range
      input
        appearance: none
        border: none
        padding: 0
        &::-webkit-slider-runnable-track
          background: var(--color-inactive)
          border-radius: var(--border-radius)
          border: var(--border)
          cursor: pointer
          height: calc(var(--height) / 3)
          width: 100%
        &::-webkit-slider-thumb
          -webkit-appearance: none
          background: var(--color-active)
          border-radius: var(--border-radius)
          border: 0
          cursor: pointer
          height: calc(var(--target-height) * 0.8)
          // margin-top: calc(((var(--height) / 3) * -1 ) - var(--border-width))
          margin-top: 0
          transform: translateY(-35%)
          width: calc(var(--height) / 3)
          &:hover
            background: var(--color-active)

    &-color
      input
        padding: 0
        &::-webkit-color-swatch-wrapper
          padding: calc(var(--base-padding) /4)
        &::-webkit-color-swatch
          border-radius: var(--border-radius)
          border: 0

    &-textarea
      textarea
        display: block
        font-family: inherit
        height: auto
        line-height: 1.3
        padding: calc(var(--base-padding) / 2) var(--base-padding) var(--base-padding) var(--base-padding)
        &::-webkit-scrollbar
          width: 0

    &-select
      .field-wrapper
        position: relative
        select
          appearance: none
          padding: 0 var(--base-padding)
        &::after
          background-color: var(--border-color)
          clip-path: polygon(50% 80%, 90% 40%, 100% 50%, 50% 100%, 0% 50%, 10% 40%)
          content: ""
          height: var(--font-size)
          pointer-events: none
          position: absolute
          right: var(--base-padding)
          top: calc(var(--base-padding) - var(--border-width))
          width: var(--font-size)
        &:hover::after
          background-color: var(--color-active)

    &-hidden
      display: none

    &-radioGroup
      .field-wrapper
        column-gap: var(--base-padding)
        display: flex
      .field-option
        cursor: pointer
        display: inline-block
        font-size: var(--font-size)
        padding: var(--base-padding) 0
        white-space: nowrap

      input[type="radio"]
        display: inline-block
        height: calc(var(--font-size) * 0.8)
        margin-right: calc(var(--base-padding) / 2)
        width: calc(var(--font-size) * 0.8)

    &-object
      legend
        border-bottom: var(--border-width) solid #d8dee9
        margin-bottom: var(--base-padding)
        padding: 0

      .fieldset
        padding-left: var(--base-padding)

        .field
          position: relative
          &:nth-child(1)::before
            height: calc(100% - (var(--base-padding) * 0.6))
            top: calc((var(--target-height) / 2 * -1) + (var(--base-padding) * 0.6))
          &::before
            border: 0 solid var(--border-color)
            border-width: 0 0 var(--border-width) var(--border-width)
            content: ""
            display: block
            height: calc(100% + var(--base-padding))
            left: calc(var(--base-padding) * -1)
            position: absolute
            top: calc((var(--target-height) / 2 * -1) - var(--base-padding))
            width: var(--base-padding)

    &-array
      ol
        counter-reset: li
        padding-left: calc(var(--base-padding) * 2)
      li
        border-left: var(--border-width) solid var(--border-color)
        display: grid
        grid-template-areas: "fieldset fieldset fieldset" ". remove remove"
        grid-template-columns: 1fr 2fr
        margin-bottom: calc(var(--base-padding))
        padding-left: var(--base-padding)
        &:before
          background-color: var(--color-inactive)
          color: var(--color-active)
          content: counter(li, decimal-leading-zero)
          counter-increment: li
          display: block
          font-size: calc(var(--base-padding) * 0.9)
          height: calc(var(--base-padding) * 2)
          line-height: calc(var(--base-padding) * 2)
          margin-left: calc(var(--base-padding) * -3)
          position: absolute
          text-align: center
          width: calc(var(--base-padding) * 2)

        &::marker
          display: none

  button
    background-color: var(--border-color)
    border-radius: var(--border-radius)
    border: var(--border-width) solid var(--border-color)
    cursor: pointer
    display: block
    font-size: var(--font-size)
    height: var(--height)
    margin-bottom: var(--base-padding)
    padding: var(--base-padding)
    transition: var(--transition-field)
    width: 100%
    &:hover
      --border-color: var(--color-hover)
    &:focus,
    &:focus-visible
      --border-color: var(--color-active)
      color: white
      outline: none
