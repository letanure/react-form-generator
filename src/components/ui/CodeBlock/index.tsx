import * as S from './styles'

export type CodeBlockProps = {
  codeString: unknown
}

const CodeBlock = ({ codeString = 'default' }: CodeBlockProps) => (
  <S.Wrapper>
    <code>
      <pre>{JSON.stringify(codeString, null, 2)}</pre>
    </code>
  </S.Wrapper>
)

export default CodeBlock
