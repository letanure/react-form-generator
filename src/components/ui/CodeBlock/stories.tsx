import { ComponentStory, ComponentMeta } from '@storybook/react'

import CodeBlock from '.'

export default {
  title: 'UI/CodeBlock',
  component: CodeBlock,
  argTypes: {}
} as ComponentMeta<typeof CodeBlock>

const codeString = `
const formConfig: FormConfig = {
  title: 'My form',
  fields: [
    {
      label: 'Placeholder text',
      placeholder: 'Placeholder test',
      name: 'namePlaceholder',
      type: 'text',
      value: ''
    },
    {
      label: 'Color',
      name: 'nameColor',
      type: 'color',
      value: '#ff0000'
    }
  ]
}
`

const Template: ComponentStory<typeof CodeBlock> = (args) => (
  <CodeBlock {...args} />
)

export const Basic = Template.bind({})
Basic.args = {
  codeString: codeString
}
