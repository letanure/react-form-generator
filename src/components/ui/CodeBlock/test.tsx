import { render } from '@testing-library/react'

import CodeBlock from '.'

const codeString = `
import { Prism as SyntaxHighlighter } from 'react-syntax-highlighter';
import { dark } from 'react-syntax-highlighter/dist/esm/styles/prism';
const Component = () => {
  const codeString = '(num) => num + 1';
  return (
    <SyntaxHighlighter language="javascript" style={dark}>
      {codeString}
    </SyntaxHighlighter>
  );
};
`

describe('<CodeBlock />', () => {
  it('should render something', () => {
    const { container } = render(<CodeBlock codeString={codeString} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
