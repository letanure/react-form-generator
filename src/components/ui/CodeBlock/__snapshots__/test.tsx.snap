// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CodeBlock /> should render something 1`] = `
.c0 {
  font-size: 11px;
  border: 1px solid #999;
  padding: 10px;
  line-height: 1.2em;
}

<div
  class="c0"
>
  <code>
    <pre>
      "\\nimport { Prism as SyntaxHighlighter } from 'react-syntax-highlighter';\\nimport { dark } from 'react-syntax-highlighter/dist/esm/styles/prism';\\nconst Component = () =&gt; {\\n  const codeString = '(num) =&gt; num + 1';\\n  return (\\n    &lt;SyntaxHighlighter language=\\"javascript\\" style={dark}&gt;\\n      {codeString}\\n    &lt;/SyntaxHighlighter&gt;\\n  );\\n};\\n"
    </pre>
  </code>
</div>
`
